jax/experimental/array_serialization/serialization.py

```diff
@@ -184,6 +185,18 @@ async def release_bytes(self, requested_bytes):
       self._cv.notify_all()
 
 
+async def transfer_shard_to_host(shard: array.Shard) -> np.ndarray:
+  data = shard.data
+  has_pinned_host = any(
+      m.kind == "pinned_host" for m in shard.device.addressable_memories())
+  if config.enable_memories.value and has_pinned_host:
+    # If available, transfer to pinned host memory
+    sharding = jax.sharding.SingleDeviceSharding(shard.device,
+        memory_kind="pinned_host")
+    data = jax.device_put(data, sharding)
+  return np.array(data, copy=False)
```

Comment:
Just return `data` instead so that we preserve the old behavior?

Reply:
Done -- do note though that this makes the function's name, `transfer_shard_to_host`, a bit misleading as nothing is transferred in the non-pinned-host case now. Fwiw, in the [follow-up PR](https://github.com/google/jax/pull/22169) we went the other way upon @cpgaffney1's suggestion, also introducing overlap for the existing path.